File: Section 2/20_GridView_Image/MainActivity.java
package com.mzu.gridviewdemo;

import android.os.Bundle;
import android.widget.GridView;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        GridView gridView = findViewById(R.id.gridView);
        gridView.setAdapter(new ImageAdapter(this));
    }
}

File: Section 3/04_Coroutines_MVVM_Retrofit_API_Kotlin/MainActivity.kt
package com.mzu.coroutinemvvm

import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.mzu.coroutinemvvm.adapter.PostAdapter
import com.mzu.coroutinemvvm.viewmodel.PostViewModel
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {
    private val postViewModel: PostViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
        val adapter = PostAdapter()
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = adapter

        postViewModel.fetchPosts()

        lifecycleScope.launch {
            postViewModel.posts.collect { postList ->
                adapter.setPosts(postList)
            }
        }
    }
}
